### ISSUE TITLE
No version for Gnome 3.30 on extensions.gnome.org

### ISSUE THREAD

**Issue by tomporter518** [aleho/gnome-shell-volume-mixer#97]
There is no version on extensions.gnome.org that can be installed for Gnome 3.30. Installing or upgrading to the apparent latest (3.32?) version will not function, at least in a Gnome 3.30 environment.

**Comment by aleho** [comment]
This extension will probably not work on 3.30 properly. I couldn't test it anyway, so sorry for that.

**Comment by tomporter518** [comment]
It works just fine if you checkout, build and install (unzip) from the v0.15.0 tag or a bit greater. I found that commit 509a874... works just fine for Gnome 3.30 (which Fedora 29 currently uses).

**Comment by aleho** [comment]
And latest master doesn't work?

I might find the time to set up a VM this week, but if you can confirm it works right now, that'd be easier ;-).

**Comment by tomporter518** [comment]
master (or at least upgrading from extensions.gnome.org which I think is master) does not work for me. Upon a reload of gnome-shell, I see these using journalctl -b:
```
Mar 25 15:44:31 zane gnome-shell[7048]: JS WARNING: [/home/tomporter/.local/share/gnome-shell/extensions/shell-volume-mixer@derhofbauer.at/lib/menu/menu.js 76]: reference to undefined property "_control"
Mar 25 15:44:31 zane gnome-shell[7048]: Extension "shell-volume-mixer@derhofbauer.at" had error: TypeError: this.options is undefined

```
Those are the only errors I see. I have begun poking around to see about debugging but I am not familiar with this extension's code base.

**Comment by tomporter518** [comment]
I don't really know anything about development in the Gnome Extension ecosystem or what may change from version to version of Gnome, but looks like constructors are never called. Nothing is initialized when the _init methods are called, resulting in the errors above.

**Comment by tomporter518** [comment]
I think I narrowed down a bit more, it seems the exception is thrown when trying to call super(); in the Menu class constructor. I added a try/catch around it and some debug logging to get this:
`ERROR | menu.js | constructor() | this.options is undefined`, as listed a few posts above.

**Comment by tingvarsson** [comment]
Latest master will probably not work due to the ES6 migration, should branch out before that for pre 3.32 if needed. Not sure how versions are handled on https://extensions.gnome.org/.

**Comment by aleho** [comment]
Yes, it's easily possible to "branch" releases and tag them to be fit for a single version only. Several releases can (and are) active for different versions of GNOME.

I'm not sure right now, but I guess GJS released with 3.30 is incompatible with master. If I remember correctly I never released for 3.30, so that'se reason there's no active installation for it on extensions.gnome.org

**Comment by tomporter518** [comment]
I initially suspected ES6 too, but Fedora 29/Gnome 3.30.x is running with Gjs 1.54.3 which includes ES6, according to its GitLab Wiki. In my testing, constructor syntax does work as well, just not in menu.js where the call to super(); throws an exception. I'm not sure if something changed in Volume.VolumeMenu, from which is derived, in Gnome 3.32...but that's where I stopped.
